internal/crypto/ptls/fips_strict_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 the Pinniped contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build fips_strict
```

Comment:
Will CI ever run these tests? I thought we didn't have FIPS-mode unit tests, if I recall correctly. We could convert this to integration tests just so they get compiled and run in CI.

Reply:
As an example, see `TestFIPSCipherSuites_Parallel` which is a unit test written in the style of an integration tests just so it always runs in CI.